### PR TITLE
[`flake8-pytest-style`] autofix for composite-assertion (PT018)

### PR DESCRIPTION
**Pull request by sbrugman**
Autofix for composite assertion/PT018

Made some choices to keep it simple, see the test file for details.

### CODE REVIEW

**Comment by spaceone** [comment]
C417 doesn't detect all cases anymore.
It fixes:

```diff
-        assert all(map(lambda v: isinstance(v, dict), iterator))
+        assert all((isinstance(v, dict) for v in iterator))
```

better would be:
```diff
-        assert all(map(lambda v: isinstance(v, dict), iterator))
+        assert all(isinstance(v, dict) for v in iterator)
```

**Comment by spaceone** [comment]
btw - replacing `filter()` in the same manner would also be helpful.

**Comment by spaceone** [comment]
with the MR the following is not detected anymore as C417: `bitstring = ''.join(map(lambda x: x in logontimes and '1' or '0', range(168)))`

**Comment by sbrugman** [comment]
Thanks, will look at them (in https://github.com/charliermarsh/ruff/pull/2693, this PR was intended for PT018)

**Comment by spaceone** [comment]
I found out why I have so many differences in the detection of that code:

Prior each finding emitted two violations:
```
C417 Unnecessary `map` usage (rewrite using a `list` comprehension)
C417 Unnecessary `map` usage (rewrite using a generator expression)
```

Now only one is detected:
```
C417 Unnecessary `map` usage (rewrite using a `list` comprehension)
```

→ I think this is ok.

File: crates/ruff/resources/test/fixtures/flake8_pytest_style/PT018.py [hunk 1]
```diff
@@ -6,13 +6,24 @@ def test_ok():
...
+    assert not something and something_else
     assert not (something or something_else)
     assert not (something or something_else or something_third)
+
+    # recursive case
+    assert not (a or not (b or c))
+    assert not (a or not (b and c))  # note that we only reduce once here
+
+    # detected, but no autofix for messages
+    assert something and something_else, "error message"
```
Comment (sbrugman) [inline crates/ruff/resources/test/fixtures/flake8_pytest_style/PT018.py:26]: Alternative: repeat message

File: docs/rules/composite-assertion.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# composite-assertion (PT018)
```
Comment (sbrugman) [inline docs/rules/composite-assertion.md:1]: Needs to be excluded after refactor

File: crates/ruff/resources/test/fixtures/flake8_pytest_style/PT018.py (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff/resources/test/fixtures/flake8_pytest_style/PT018.py:26]: I think this is the right call (not to fix these cases).

**Pull request merged**